server/api/user/update/controller.test.js

```diff
@@ -0,0 +1,511 @@
+      ok: false,
+      error: {
+        code: 10005,
+        message: `Email address already in use`,
+      },
+    });
+  });
+
+  test('returns error when trying to add an org to the user', async () => {
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        orgs: ['5b2d5dd0cd86b77258e16d39'], //random id
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: false,
+      error: {
+        code: 400,
+        message: 'Invalid request body',
+      },
+    });
+  });
+
+  test('updates user with new image if picture url is provided', async () => {
+    const newPictureURL = 'https://www.acme.com/pictures/v2/coyote.png';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: newPictureURL,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: newPictureURL,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+
+  test('removes image from user if picture property is null', async () => {
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: null,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: null,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+
+  describe('Requestor is superuser', () => {
+    test('returns valid user when setting emails as verified', async () => {
+      const emails = [...unauthorisedUser.emails, {
+        address: 'not@verified.com',
+        isVerified: true,
+        isPrimary: false,
+      }];
+      const res = await request(server)
+        .post('/api/user.update')
+        .set('Authorization', `Bearer ${superuserSession.accessToken}`)
+        .send({
+          id: unauthorisedUser.id,
+          emails,
+        });
+      expect(res.status).toBe(200);
+      expect(res.body).toMatchObject({
+        ok: true,
+        user: {
+          id: unauthorisedUser.id,
+          emails: expect.arrayContaining([{
+            address: expect.any(String),
+            isVerified: true,
+            isPrimary: expect.any(Boolean),
+          }]),
+        },
+      });
+    });
+    // discuss if we need to explicitly make this case work
+    xtest('returns valid user when setting non-verified as primary', async () => {
```

Comment:
This test is here in case we want to refactor the ability of a superuser to add as `isPrimary` a non-verified email. It might also create a bug in the future for us, since we are always expecting the `isPrimary` email to also has an `isVerified` property set to true.

Reply:
This will never happen however, so feel free to remove.